src/js/actions/export.js

```diff
@@ -317,6 +326,50 @@ define(function (require, exports) {
     setLayerExportEnabled.writes = [locks.JS_DOC, locks.PS_DOC];
 
     /**
+     * Sets the exportEnabled flag for all artboards
+     * @private
+     * @param {Document} document Owner document
+     * @param {boolean=} exportEnabled
+     * @return {Promise}
+     */
+    var setAllArtboardsExportEnabled = function (document, exportEnabled) {
+        var documentExports = this.flux.stores.export.getDocumentExports(document.id);
+
+        if (!documentExports) {
+            return Promise.resolve();
+        }
+
+        var layersWithExports = documentExports.getLayersWithExports(document, true);
+
+        return this.transfer(setLayerExportEnabled, document, layersWithExports, exportEnabled);
+    };
+    setAllArtboardsExportEnabled.reads = [locks.JS_DOC];
+    setAllArtboardsExportEnabled.writes = [];
+    setAllArtboardsExportEnabled.transfers = [setLayerExportEnabled];
+
+    /**
+     * Sets the exportEnabled flag for all layers that have at least one configured asset
+     * @private
+     * @param {Document} document Owner document
+     * @param {boolean=} exportEnabled
+     * @return {Promise}
+     */
+    var setAllNonABLayersExportEnabled = function (document, exportEnabled) {
+        var documentExports = this.flux.stores.export.getDocumentExports(document.id);
+
+        if (!documentExports) {
+            return Promise.resolve();
+        }
+
+        var layersWithExports = documentExports.getLayersWithExports(document, false);
+
+        return this.transfer(setLayerExportEnabled, document, layersWithExports, exportEnabled);
+    };
+    setAllNonABLayersExportEnabled.reads = [locks.JS_DOC];
```

Comment:
My vague feeling is that this won't buy us much, since there is a high coupling between the two stores.  BUT, I'll give it a shot because it might increase clarity... and I should probably let the machine decide if the more granular lock helps, instead of my intuition.